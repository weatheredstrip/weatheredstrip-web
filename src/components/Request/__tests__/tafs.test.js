import React from 'react'
import { render, cleanup } from '@testing-library/react'
import Tafs from '../tafs.react'
import '@testing-library/jest-dom/extend-expect'
import testData from './test_data.json'
afterEach(cleanup)

const airports = Object.keys(testData)

airports.forEach(airport => {
  it('taf renders properly', () => {
    const { container } = render(<Tafs data={testData[airport].taf} />)
    const tafs = container.firstChild
    expect(tafs).toMatchSnapshot()
  })
})

it('empty taf renders empty', () => {
  const { container } = render(<Tafs data={[]} />)
  const tafs = container.firstChild
  expect(tafs).toMatchSnapshot()
})

it('no taf renders empty', () => {
  const { container } = render(<Tafs />)
  const tafs = container.firstChild
  expect(tafs).toMatchSnapshot()
})
