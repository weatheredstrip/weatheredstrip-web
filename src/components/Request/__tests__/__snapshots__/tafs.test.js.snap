// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`empty taf renders empty 1`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;

exports[`no taf renders empty 1`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;

exports[`taf renders properly 1`] = `
<div
  class="tafs"
>
  <div
    class="notif-text"
  >
    TAF CYMX 251138Z 2512/2612 VRB03KT P6SM SCT180 BKN220 TEMPO 2512/2513 1/2SM FG BKN001 OVC180
  </div>
  <div
    class="notif-text"
  >
    FM251500 24005KT P6SM BKN160 BKN240
  </div>
  <div
    class="notif-text"
  >
    FM260200 26005KT P6SM SKC
  </div>
  <div
    class="notif-text"
  >
    FM260600 VRB03KT P6SM BKN040
  </div>
  <div
    class="notif-text"
  >
    PROB30 2606/2612Â 1/2SM FG BKN001
  </div>
  <div
    class="notif-text"
  >
    RMK NXT FCST BY 251800Z
  </div>
</div>
`;

exports[`taf renders properly 2`] = `
<div
  class="tafs"
>
  <div
    class="notif-text"
  >
    KSFO 251504Z 2515/2618 00000KT P6SM SKC
  </div>
  <div
    class="notif-text"
  >
    FM251600 10003KT 6SM FU FEW070
  </div>
  <div
    class="notif-text"
  >
    FM252000 04003KT 6SM FU FEW070
  </div>
  <div
    class="notif-text"
  >
    FM260000 29012KT 6SM FU FEW070
  </div>
  <div
    class="notif-text"
  >
    FM260400 28008KT 6SM FU FEW080
  </div>
</div>
`;

exports[`taf renders properly 3`] = `
<div
  class="tafs"
>
  <div
    class="notif-text"
  >
    TAF EGPK 251102Z 2512/2612 19007KT 9999 FEW020 SCT035
  </div>
  <div
    class="notif-text"
  >
    PROB30 TEMPO 2512/2518 6000 RA BKN012 BECMG 2609/2612 25012KT
  </div>
</div>
`;

exports[`taf renders properly 4`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;

exports[`taf renders properly 5`] = `
<div
  class="tafs"
>
  <div>
    No TAF is issued for this station.
  </div>
</div>
`;
